### DISCUSSION TITLE
Help understanding how to program the ATMega32u4

### DISCUSSION THREAD

**Discussion by ceasardaking** [xyphro/UsbGpib#62] [Q&A]
Hi, i'm having trouble understanding on flashing the bin file that is for test and measurement. I'm using Microchip Studio and i'm able to detect the chip in it's default USB DFU mode. Once i get into the memory and flash the bin file. i try to run it on an instrument and checking it with NI Max and i don;t see it. i also can't get back into DFU mode to reprogram it. I have a hard time using the USBTiny with Microchip Studio. Will someone be able to help me guide me on installing the software properly. I can restart on a fresh dfu chip.

**Comment by xyphro** [comment]
Hi Caesar, 

don't use DFU mode, you have to program it using a real AVR programmer. The HW page contains instructions on how to program.

Best regards,

Kai
